crypto/core_algorithm.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/core.h>
+#include "internal/core.h"
+#include "internal/property.h"
+#include "internal/provider.h"
```

Comment:
Sure.  I found that `EVP_MAC_do_all` would do very much what `EVP_MAC_fetch`, so instead of creating something that I knew would be copied to any other API where we care about presentation, I'd make something that's general purpose enough to code the major part of the job and make the like of `EVP_MAC_do_all` as well as `EVP_MAC_fetch` easy to write.

This needs documenting, btw...

Reply:
Ah. Thanks for the explanation.